bin/basket.js

```diff
@@ -14,88 +14,115 @@ var config = require('../config').getProperties()
-                op: 'basketRequest',
-                uid: message.uid,
-                locale: message.locale,
-                err: err
-              }
-            )
-            return
-          }
-          // Log at error level for HTTP-level errors, info level otherwise.
-          if (res.statusCode < 200 || res.statusCode >= 300) {
-            log.error(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                uid: message.uid,
-                locale: message.locale,
-                body: body
-              }
-            )
-          } else {
-            log.info(
-              {
-                op: 'basketRequest',
-                status: res.statusCode,
-                locale: message.locale,
-                body: body
-              }
-            )
-          }
-          message.del()
-        }
-      )
-    }
-    else {
-      message.del()
+
+    var messageHandler = messageHandlers[message.event]
+    if (messageHandler) {
+      return messageHandler(message)
     }
+
+    message.del()
   }
 )
 basketQueue.start()
+
+function onVerified (message) {
+  forwardEvent(message, '/fxa-register/', {
+    fxa_id: message.uid,
+    email: message.email,
+    // Basket won't accept empty or null `accept_lang` field,
+    // so we default to en-US.  This should only happen if
+    // the user has not sent an explicit Accept-Language header.
+    accept_lang: message.locale || 'en-US'
+  }, function (data) {
+    data.uid = message.uid
+    data.locale = message.locale
+    return data
+  })
+}
+
+function forwardEvent (message, endpoint, data, decorateLog) {
+  // Ignore email addresses that are clearly from dev testing.
+  if (shouldIgnoreEmail(message.email)) {
+    message.del()
+    return
+  }
+
+  // Forward all others to basket API.
+  request({
+    url: config.basket.apiUrl + endpoint,
+    strictSSL: true,
+    method: 'POST',
+    headers: {
+      'X-API-Key': config.basket.apiKey
+    },
+    form: data,
+  },
+  function (err, res, body) {
+    // Log and retry for network-level errors.
+    if (err) {
+      log.error(decorateLog({
+        op: 'basketRequest',
+        err: err,
+        endpoint: endpoint
+      }))
+      return
+    }
+
+    // Log at error level for HTTP-level errors, info level otherwise.
+    if (res.statusCode < 200 || res.statusCode >= 300) {
+      log.error(decorateLog({
+        op: 'basketRequest',
+        status: res.statusCode,
+        body: body,
+        endpoint: endpoint
+      }))
+    } else {
+      log.info(decorateLog({
+        op: 'basketRequest',
+        status: res.statusCode,
+        body: body,
+        endpoint: endpoint
+      }))
+    }
+    message.del()
+  })
+}
+
+function shouldIgnoreEmail (email) {
+  if (email.match(/@restmail.net$/)) {
+    return true
+  }
+
+  if (email.match(/@restmail.lcip.org$/)) {
+    return true
+  }
+
+  return false
+}
+
+function onLogin (message) {
```

Comment:
nit: IMO this should be up near its `onVerified` counterpart rather than having two similar function separated by a bunch of helpers.